src/coreclr/vm/stackwalk.cpp

```diff
@@ -290,6 +290,45 @@ PTR_VOID CrawlFrame::GetExactGenericArgsToken()
     }
 }
 
+void CrawlFrame::InitializeExactGenericInstantiations()
+{
+    CONTRACTL {
+        SUPPORTS_DAC;
+        NOTHROW;
+        GC_NOTRIGGER;
+    } CONTRACTL_END;
+
+    // Turn into cooperative GC mode
+    GCX_COOP();
+
+    if (pFunc != NULL && pFunc->HasClassOrMethodInstantiation() && pFunc->IsSharedByGenericInstantiations())
+    {
+        // Get exact instantiations for shared generics where possible
+        PTR_VOID pExactGenericArgsToken = GetExactGenericArgsToken();
+
+        if (pExactGenericArgsToken != NULL)
+        {
+            TypeHandle th;
+            MethodDesc* pConstructedFunc = NULL;
+            if (Generics::GetExactInstantiationsOfMethodAndItsClassFromCallInformation(pFunc, pExactGenericArgsToken, &th, &pConstructedFunc))
+            {
+#ifndef DACCESS_COMPILE
+                if (pConstructedFunc->IsSharedByGenericInstantiations())
+                {
+                    InstantiatedMethodDesc *pInstMD = InstantiatedMethodDesc::FindLoadedInstantiatedMethodDesc(th.GetMethodTable(), pConstructedFunc->GetMemberDef(), Instantiation(), false);
+
+                    if (pInstMD != NULL)
+                    {
+                        pConstructedFunc = pInstMD;
+                    }
```

Comment:
It should really be
```cpp
pConstructedFunc = InstantiatedMethodDesc::FindOrCreateExactClassMethod(th.GetMethodTable(), pConstructedFunc);
```
but this would possibly throw or trigger a GC.